### DISCUSSION TITLE
Safari problem with first example on Start page

### DISCUSSION THREAD

**Discussion by garyrob** [alpinejs/alpine#2216] [7. Bugs (archived)]
On https://alpinejs.dev/start-here, I make an html file with the code in the first example. The text says, "Open your file in a web browser, if you see I ❤️ Alpine, you're ready to rumble!"  If I open the file in Firefox, it looks great. But if I open it in Safari (14.1.2), I see gibberish characters where the heart should be.

It's due to something unusual about Safari's rendering, I assume, but it's kind of an issue for your tutorial when you actually tell the user that all is good if they see something they won't see if they happen to be using Safari!

**Comment by p-syche** [comment]
Hi! I see this was asked a year and a half ago, but it got me checking down the rabbit hole of html entities and emojis :D
So here's what's happening: many browsers are capable of taking the emoji in your code and converting it to whatever they need (like UTF-8) in order to display it. Apparently safari has a problem with that, so it displays "gibberish":
<img width="1369" alt="Zrzut ekranu 2023-05-18 o 10 47 27" src="https://github.com/alpinejs/alpine/assets/4152181/7eeb553e-f9d9-4f50-8f8f-2d6b0c82b4d1">

while other browsers are fine with displaying ❤️

What you want to do is use HTML entities - so that you have full control over what is displayed. You can find lists of emojis and their symbols here for example: https://www.htmlsymbols.xyz/heart-symbols
HTML Entity for the red heart emoji above is: `&#10084;`

<img width="961" alt="Zrzut ekranu 2023-05-18 o 10 51 06" src="https://github.com/alpinejs/alpine/assets/4152181/0debcc04-e6ca-4d3f-ae53-638ad7758308">

At this point you may wonder - how come the description of this emoji is "heavy black heart"? weird, right?
And what if you open this emoji list in Safari? Boom! Heavy black heart!
<img width="1355" alt="Zrzut ekranu 2023-05-18 o 10 54 01" src="https://github.com/alpinejs/alpine/assets/4152181/56dd930a-9b4f-48f8-a887-1ebae26ba4d4">
So let's change the code to use the correct HTML Entity:
```
<html>
  <head>
    <script
      defer
      src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"
    ></script>
  </head>
  <body>
    <h1
      x-data="{ message: 'I &#10084; Alpine' }"
      x-text="message"
    ></h1>
  </body>
</html>
```
Safari:
<img width="1363" alt="Zrzut ekranu 2023-05-18 o 10 51 57" src="https://github.com/alpinejs/alpine/assets/4152181/fdc43efe-6320-4c73-bd0a-ac13888143be">

Firefox 🤯 
<img width="470" alt="Zrzut ekranu 2023-05-18 o 10 51 38" src="https://github.com/alpinejs/alpine/assets/4152181/49030f00-5a4c-40ed-80b2-fdbbdd25aaaa">

Here's safari with a couple more emojis passed as HTML entities:


<img width="694" alt="Zrzut ekranu 2023-05-18 o 10 52 30" src="https://github.com/alpinejs/alpine/assets/4152181/325c130c-0b68-4e5e-a82d-b9216f3349b3">


As you can see Safari is perfectly capable of displaying emojis. It even displays correct-er ones than Firefox 😬 

I have no idea if this helps you at all @garyrob , but I hope you had fun with me on this adventure :)

**Reply by ekwoka** [reply]
Interesting!

It would be like Safari to be different.

**Comment by p-syche** [comment]
P.S.

This comment inclued the heart emoji. If you open it on safari you will see a red heart emoji ❤️

But it's not _actually_ an emoji 😆  it's a fallback image, taken from here:

https://github.githubassets.com/images/icons/emoji/unicode/2764.png
